### PR TITLE
[Bug-9608] Serialize the task definition failed

### LINKED ISSUE #9608: [Bug] [DAO] Due to the task param map, serialize the task definition failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

TaskDefinition.java

When the local params is null or the property value is null in the local param list , it will cause the `NullPointException` in the serialization of the task definition. Because the `jackson` json lib  will use the method `getTaskParamMap` to serialize the object;

### What you expected to happen

When the local params is null or the property value is null in the local param list , the task definition object could be serialized normally .  The `taskParamMap` could be empty HashMap or the map contains null value;

### How to reproduce

Case 1 :  The `taskParams` doesn't contain `localParams` parameter , which is null;

Case 2:  The `taskParams` contains `localParams` parameter  , whose property element has a null value;

### Anything else

_No response_

### Version

2.0.5

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by lipandongDr**
Fix: #9608

### CODE REVIEW

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskDefinition.java [hunk 1]
```diff
@@ -309,9 +309,18 @@ public void setTaskParamMap(Map<String, String> taskParamMap) {
...
-            if (localParams != null) {
+
+            //If a jsonNode is null, not only use !=null, but also it should use the isNull method to be estimated.
+            if (localParams != null && !localParams.isNull()) {
                 List<Property> propList = JSONUtils.toList(localParams.toString(), Property.class);
-                taskParamMap = propList.stream().collect(Collectors.toMap(Property::getProp, Property::getValue));
+
+                if (CollectionUtils.isNotEmpty(propList)) {
+                    taskParamMap = new HashMap<>();
+                    // Don't use java 8 stream to convert map, because the value will probably null,and it will cause the NullPointException.
```
Comment (SbloodyS) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskDefinition.java]: I think this comment is unnecessary. It should be removed.

Comment (naziD) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskDefinition.java:319]: > 
It should be emphasized that the latter deployer don't use stream to convert the map . The key reason of this issue is caused by it .

Comment (lipandongDr) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskDefinition.java]: > I think this comment is unnecessary. It should be removed.

It's done

**Review by caishunfeng — APPROVED** [review]
LGTM if CI pass

**Comment by caishunfeng** [comment]
Thanks for your contribution @naziD

**Pull request merged**